File: chap1/julia/code3.jl
using Distributions
using GLMNet

include("myFunc.jl")
using .MyFunc

## データの生成(d=100, rho=0.3とした場合)
# 変数の設定
n = 100
d = 100
ρ = 0.3
s = Int(d/2)

β = zeros(d)
β[1:s] .= 2

# エラスティックネットのパラメータ
λ = 4 * sqrt(2 * log(d)/n)
α = 0.5

# 説明変数ベクトルの平均と共分散行列の作成
m = zeros(d)

tmp = MyFunc.outer(1:d, 1:d, -)
Σ = ρ .^ abs.(tmp)

x = rand(MvNormal(m, Σ), n)
y = x * β .+ rand(Normal(), n)

## エラスティックネットによるパラメータの推定
enetFit = glmnet(x, y, lambda=[λ], alpha=α)

File: chap1/R/code3.r
# パッケージの呼び出し
library(glmnet)
library(MASS)

# データの生成(d=100, rho=0.3とした場合)
## 変数の設定
n <- 100; d<-100; rho <- 0.3;
s <- d/2

beta <- rep(0, d); beta[1:s] <- 2

## エラスティックネットのパラメータ
lambda <- 4 * sqrt(2 * log(d)/n)
alpha <- 0.5

## 説明変数ベクトルの平均と共分散行列の作成
m <- rep(0, d)
tmp <- outer(1:d, 1:d, "-")
Sigma <- rho^{abs(tmp)}

x <- mvrnorm(n, m, Sigma)
y <- x %*% beta + rnorm(n, 0, 1)

# エラスティックネットによるパラメータの推定
enet.fit <- glmnet(x, y, family="gaussian", lambda = lambda, alpha = alpha)
enet.fit